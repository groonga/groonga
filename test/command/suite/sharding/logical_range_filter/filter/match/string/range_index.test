#@on-error omit
plugin_register sharding
#@on-error default

table_create Terms TABLE_PAT_KEY ShortText \
  --default_tokenizer TokenBigram

table_create Logs_20150205 TABLE_NO_KEY
column_create Logs_20150205 timestamp COLUMN_SCALAR Time
column_create Logs_20150205 memo COLUMN_SCALAR ShortText
column_create Logs_20150205 action COLUMN_SCALAR Text

table_create Timestamp_20150205 TABLE_PAT_KEY Time
column_create Timestamp_20150205 index COLUMN_INDEX Logs_20150205 timestamp

column_create Terms index_20150205 COLUMN_INDEX|WITH_POSITION|WITH_SECTION \
  Logs_20150205 memo,action


load --table Logs_20150205
[
{
  "timestamp": "2015-02-05 13:49:00",
       "memo": "2015-02-05 13:49:00",
     "action": "Start"
},
{
  "timestamp": "2015-02-05 13:50:00",
       "memo": "2015-02-05 13:50:00",
     "action": "Running"
},
{
  "timestamp": "2015-02-05 13:51:00",
       "memo": "2015-02-05 13:51:00",
     "action": "Running"
},
{
  "timestamp": "2015-02-05 13:52:00",
       "memo": "2015-02-05 13:52:00",
     "action": "Shutdown"
}
]

#@add-important-log-levels debug
log_level --level debug
#@collect-query-log true
logical_range_filter Logs timestamp \
  --limit 1 \
  --filter 'action @ "nn"'
#@collect-query-log false
log_level --level notice
#@remove-important-log-levels debug
