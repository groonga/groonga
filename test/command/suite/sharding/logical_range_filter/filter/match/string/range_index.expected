plugin_register sharding
[[0,0.0,0.0],true]
table_create Terms TABLE_PAT_KEY ShortText   --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
table_create Logs_20150205 TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Logs_20150205 timestamp COLUMN_SCALAR Time
[[0,0.0,0.0],true]
column_create Logs_20150205 memo COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Logs_20150205 action COLUMN_SCALAR Text
[[0,0.0,0.0],true]
table_create Timestamp_20150205 TABLE_PAT_KEY Time
[[0,0.0,0.0],true]
column_create Timestamp_20150205 index COLUMN_INDEX Logs_20150205 timestamp
[[0,0.0,0.0],true]
column_create Terms index_20150205 COLUMN_INDEX|WITH_POSITION|WITH_SECTION   Logs_20150205 memo,action
[[0,0.0,0.0],true]
load --table Logs_20150205
[
{
  "timestamp": "2015-02-05 13:49:00",
       "memo": "2015-02-05 13:49:00",
     "action": "Start"
},
{
  "timestamp": "2015-02-05 13:50:00",
       "memo": "2015-02-05 13:50:00",
     "action": "Running"
},
{
  "timestamp": "2015-02-05 13:51:00",
       "memo": "2015-02-05 13:51:00",
     "action": "Running"
},
{
  "timestamp": "2015-02-05 13:52:00",
       "memo": "2015-02-05 13:52:00",
     "action": "Shutdown"
}
]
[[0,0.0,0.0],4]
log_level --level debug
[[0,0.0,0.0],true]
logical_range_filter Logs timestamp   --limit 1   --filter 'action @ "nn"'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        "action",
        "Text"
      ],
      [
        "memo",
        "ShortText"
      ],
      [
        "timestamp",
        "Time"
      ]
    ],
    [
      "Running",
      "2015-02-05 13:50:00",
      1423111800.0
    ]
  ]
]
#|d| [logical_range_filter][range-index] <Logs_20150205>: hit ratio (1.5=6/4) >= threshold (0.2)
#>logical_range_filter --filter "action @ \"nn\"" --limit "1" --logical_table "Logs" --shard_key "timestamp"
#:000000000000000 filter(1)[Logs_20150205]: action @ "nn"
#:000000000000000 output(1)
#<000000000000000 rc=0
log_level --level notice
[[0,0.0,0.0],true]
